core/db/migrate/20240821173254_create_spree_permission_sets_in_core.rb

```diff
@@ -0,0 +1,11 @@
+class CreateSpreePermissionSetsInCore < ActiveRecord::Migration[7.0]
+  def change
+    unless table_exists? :spree_permission_sets
```

Comment:
This check might interfere with the automatic `up/down` for the migration, we should rather switch to use those explicitly or maybe use the `if_not_exist: true` option, assuming it's available on all the rails versions/dbs we support.

https://api.rubyonrails.org/classes/ActiveRecord/ConnectionAdapters/SchemaStatements.html#method-i-create_table

Reply:
I was wondering about this actually, my first intuition would be that it would prevent the rollback with the change, but I wasn't sure about the degree to which the `change` method is fancy rails magic, so it's probably better to be safe than sorry!

Do we run our CI specs against all supported versions and DBs? (eg can I rely on the test suite to flag this `if_not_exist: true` as unavailable for our supported versions or should I cross check that elsewhere?)

Reply:
We should use `if_not_exist: true`, which is supported since Rails 6.1 and we support Rails 7.0 and above, and remove the `unless` check here.

Reply:
Oops sorry, it looks like I added the `if_not_exist: true` but forgot to remove the `unless` check 😓